### PR TITLE
Added colorization for tuple types, paren types. 

### PR DESCRIPTION
**Pull request by rbuckton**
Added colorization for tuple types, paren types. Improved colorization for cast and type names.

### CODE REVIEW

File: TypeScript.YAML-tmLanguage [hunk 1]
```diff
@@ -200,50 +200,78 @@ repository:
...
+    end: '\)'
     patterns:
-    - include: '#comment'
+    - include: '#type-or-function-type-parameter'
     - include: '#type'
 
+  type-or-function-type-parameter:
+    name: meta.type.paren.cover.ts
+    match: '(\.\.\.)?\s*([a-zA-Z_$][\w$]*)\s*(\?)?\s*(:)?\s*(?=,|\)\s*=>)'
+    captures:
```
Comment (billti) [inline TypeScript.YAML-tmLanguage]: What is the first capture group for?  Appears to optionally match any 3 chars.

**Comment by billti** [comment]
Thanks Ron! This looks a little outdated and can't be merged currently.  Is this still something you want to get in?

File: TypeScript.YAML-tmLanguage (hunk 1 above)
Comment (rbuckton) [inline TypeScript.YAML-tmLanguage]: The first capture group captures the literal `...` which we use for rest arguments, which is why it is escaped as `\.`.

**Comment by zhengbli** [comment]
This PR is outdated, therefore I incorporated most of the changes to PR #155. Closing this one.

**Pull request closed without merging**